### ISSUE TITLE
macOS 10.11 with Xcode 8: Symbol not found: _clock_gettime

### ISSUE THREAD

**Issue by ilovezfs** [tarantool/tarantool#1777]
If built with Xcode 8 on macOS 10.11 (El Capitan), tarantool (1.7 branch and 1.8 branch) fails at run time with the following error:
```
bash-3.2$ /usr/local/Cellar/tarantool/1.7.1-394/bin/tarantool /tmp/tarantool-test-20160919-54314-1ed5e4z/test.lua
dyld: lazy symbol binding failed: Symbol not found: _clock_gettime
  Referenced from: /usr/local/Cellar/tarantool/1.7.1-394/bin/tarantool
  Expected in: /usr/lib/libSystem.B.dylib

dyld: Symbol not found: _clock_gettime
  Referenced from: /usr/local/Cellar/tarantool/1.7.1-394/bin/tarantool
  Expected in: /usr/lib/libSystem.B.dylib

Trace/BPT trap: 5
```

I'm using the following workaround for Homebrew until there's an upstream fix:
```
    if MacOS.version == "10.12" && MacOS::Xcode.installed? && MacOS::Xcode.version >= "8.0"
      args << "-DHAVE_CLOCK_GETTIME:INTERNAL=0"

      inreplace "src/trivia/util.h", "#ifndef HAVE_CLOCK_GETTIME",
                                     "#ifdef UNDEFINED_GIBBERISH"
    end
```

**Comment by rtsisyk** [comment]
Please paste output of cmake related to clock_gettime().

**Comment by ilovezfs** [comment]
```
Determining if the function clock_gettime exists passed with the following output:
Change Dir: /tmp/tarantool-20160920-21540-1g36tpx/tarantool-1.7.1.401/CMakeFiles/CMakeTmp

Run Build Command:"/usr/local/Homebrew/Library/Homebrew/shims/super/gmake" "cmTC_ad457/fast"
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f CMakeFiles/cmTC_ad457.dir/build.make CMakeFiles/cmTC_ad457.dir/build
Building C object CMakeFiles/cmTC_ad457.dir/CheckFunctionExists.c.o
/usr/local/Homebrew/Library/Homebrew/shims/super/clang    -fexceptions -funwind-tables -fno-common -fopenmp -msse2 -DCHECK_FUNCTION_EXISTS=clock_gettime -mavx -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk -mmacosx-version-min=10.11   -o CMakeFiles/cmTC_ad457.dir/CheckFunctionExists.c.o   -c /usr/local/Cellar/cmake/3.6.2/share/cmake/Modules/CheckFunctionExists.c
Linking C executable cmTC_ad457
/usr/local/Cellar/cmake/3.6.2/bin/cmake -E cmake_link_script CMakeFiles/cmTC_ad457.dir/link.txt --verbose=1
/usr/local/Homebrew/Library/Homebrew/shims/super/clang   -fexceptions -funwind-tables -fno-common -fopenmp -msse2 -DCHECK_FUNCTION_EXISTS=clock_gettime -mavx -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk -mmacosx-version-min=10.11 -Wl,-search_paths_first -Wl,-headerpad_max_install_names   CMakeFiles/cmTC_ad457.dir/CheckFunctionExists.c.o  -o cmTC_ad457
```

**Comment by ilovezfs** [comment]
The underlying cause is the new SDK's weak symbols:
```
iMac-TMP:SDKs joe$ pwd
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs
iMac-TMP:SDKs joe$ grep 'weak$os10.11' MacOSX.sdk/usr/lib/system/libsystem_c.tbd
    symbols:         [ '$ld$weak$os10.11$_basename_r', '$ld$weak$os10.11$_clock_getres', 
                       '$ld$weak$os10.11$_clock_gettime', '$ld$weak$os10.11$_clock_settime', 
                       '$ld$weak$os10.11$_dirname_r', '$ld$weak$os10.11$_getentropy', 
                       '$ld$weak$os10.11$_mkostemp', '$ld$weak$os10.11$_mkostemps', 
iMac-TMP:SDKs joe$
```

```
iMac-TMP:SDKs joe$ grep clock_gettime MacOSX.sdk/usr/include/time.h 
int clock_gettime(clockid_t __clock_id, struct timespec *__tp);
__uint64_t clock_gettime_nsec_np(clockid_t __clock_id);
```

So it's defined, but you won't find that out at build time unless you pass `-no_weak_imports` in LDFLAGS. But you can't easily do that because that flag is not supported on earlier versions of Xcode, including the CLT 7.3.1 which is the latest available for 10.11. So even if Xcode 8 is installed, using `-no_weak_imports` can still fail.

But assuming you don't pass `-no_weak_imports` it will find the symbol is defined in time.h and compile. But then crash at run time. So the most reliable check is that it both builds and executes, since simply successfully compiling does not mean you actually have the symbol available at runtime anymore.
